docs/pages/tutorials/minimal/use-onchain.mdx

```diff
@@ -0,0 +1,255 @@
+
+1. Download the source code and install the necessary packages.
+
+   ```sh copy
+   git clone https://github.com/latticexyz/mud.git
+   cd mud/examples/onchain-client/
+   pnpm install
+   ```
+
+1. Set the necessary environment variables.
+
+   ```sh copy
+   . .env
+   ```
+
+   This file specifies two variables:
+
+   | Variable    | Meaning                                                                                                                 |
+   | ----------- | ----------------------------------------------------------------------------------------------------------------------- |
+   | PRIVATE_KEY | A private key that has ETH on the blockchain. This is a different private key than the one that deployed the `World`.   |
+   | WORLD_ADDR  | The address of the `World` contract. The value in `.env` is the initial world deployed by `pnpm dev`, change as needed. |
+
+1. Deploy the sample contact.
+
+   ```sh copy
+   forge create OnchainClient --private-key $PRIVATE_KEY --constructor-args $WORLD_ADDR
+   ```
+
+1. Set `CLIENT_ADDR` to the address to which the contract is deployed.
+
+1. Get the value of `Counter` from the client.
+
+   ```sh copy
+   cast call $CLIENT_ADDR "readCounter()"
+   ```
+
+1. Increment `Counter` and read the value again.
+   See it is one more.
+
+   ```sh copy
+   cast send --private-key $PRIVATE_KEY $CLIENT_ADDR "add1()"
+   cast call $CLIENT_ADDR "readCounter()"
+   ```
+
+1. Attempt to reset the counter, which bypasses the `System` logic and modifies `Counter` directly.
+   See that this attempt fails.
+
+   ```sh copy
+   cast send --private-key $PRIVATE_KEY $CLIENT_ADDR "reset()"
+   ```
+
+   The error message should be similar to:
+
+   ```
+   (code: 3, message: execution reverted, data: Some(String("0x")))
+   ```
+
+<details>
+
+<summary>Explanation of `src/OnchainClient.sol`</summary>
+
+Let's go over the contract source code (`src/OnchainClient.sol`).
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { Counter } from "codegen/Tables.sol";
+import { IWorld } from "codegen/world/IWorld.sol";
+```
+
+We import `Counter` and `IWorld` from the code that was generated using `mud codegen`.
+The entire `src/codegen` directory is copied [from the template](https://github.com/latticexyz/mud/tree/main/templates/vanilla/packages/contracts/src/codegen) into the new application's `lib` directory.
+
+```solidity
+contract OnchainClient {
+  IWorld immutable world;
+
+  constructor(address world_) {
+    world = IWorld(world_);
+  }
+```
+
+Define a variable to hold the address of the `World`.
+The constructor receives the value for it.
+
+```solidity
+function readCounter() public view returns (uint32) {
+  return Counter.get(world);
+}
+```
+
+To get the value of the counter we use the `Counter` library we imported from `codegen/Tables.sol`.
+There are two versions of `get` in that library.
```

Comment:
How was this table created? I think it would be helpful if the example project included a `mud.config.ts` to generate this or other table libraries